### PR TITLE
New Custom Card : alarm time and toggle

### PR DESCRIPTION
**Pull request by benbur98**
A new custom card that can be used to set the time and turn on/off an Alarm.
This brings together an input datetime helper and boolean helper.

I built this on top of @sildehoop 's custom_card_input_datetime, however, I couldn't add this boolean functionality without several breaking changes to that card - so a new one is required.
image
image
image

- This PR fixes or closes issue: N/A
- This PR is related to issue: N/A
- Link to documentation pull request: N/A

## Checklist
- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [X] I have followed the [contribution guidelines](https://github.com/UI-Lovelace-Minimalist/UI/blob/main/.github/CONTRIBUTING.md)
    - [X] This PR is for a custom-card or documentation change and therefore directed to the `main` branch.
    - [ ] This PR is for a official card or any other directly to the integration related change and therefore directed to the `release` branch.

### CODE REVIEW

**Comment by benbur98** [comment]
## Remaining Tasks
- [ ] Fix the Force Background Color - same as the new light/cover/fan cards

**Comment by basbruss** [comment]
> ## Remaining Tasks
> * [ ]  Fix the Force Background Color - same as the new light/cover/fan cards

Any update on this?

**Comment by benbur98** [comment]
I've had no time to jump back into this for a while unfortunately. Hopefully can have it complete in the next week though.

**Comment by benbur98** [comment]
Thus is now complete, ready to merge.

File: custom_cards/custom_card_alarm_time/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Author: sildehoop - 2022
+Version: 1.0.0
+
+## Changelog
+
+<details>
+<summary>1.0.0</summary>
+Initial Release
+</details>
+
```
Comment (basbruss) [inline custom_cards/custom_card_alarm_time/README.md:27]: Missing table with configurable variables and default values

File: custom_cards/custom_card_alarm_time/card_alarm_time.yaml [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                      var seconds = "0" + date.getSeconds();
+                      var formattedTime = hours + ':' + minutes.substr(-2) + ':' + seconds.substr(-2);
+                      return formattedTime;
+                    ]]]
+              icon: "mdi:minus"
+              styles:
+                card:
+                  - background-color: >
+                      [[[
+                        if (!hass.themes.darkMode && variables.mui_card_alarm_time_force_background_color) {
```
Comment (basbruss) [inline custom_cards/custom_card_alarm_time/card_alarm_time.yaml:200]:
```suggestion
                        if (!hass.themes.darkMode && variables.ulm_card_alarm_time_force_background_color) {
```

File: custom_cards/custom_card_alarm_time/card_alarm_time.yaml [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+                          if (entity.state != "off") {
+                            return 'rgba(250,250,250,0.8)';
+                          }
+                        }
+                        return 'rgba(var(--color-theme),0.05)';
+                      ]]]
+                icon:
+                  - color: >
+                      [[[
+                        if (!hass.themes.darkMode && variables.mui_card_alarm_time_force_background_color) {
```
Comment (basbruss) [inline custom_cards/custom_card_alarm_time/card_alarm_time.yaml:210]:
```suggestion
                        if (!hass.themes.darkMode && variables.ulm_card_alarm_time_force_background_color) {
```

File: custom_cards/custom_card_alarm_time/card_alarm_time.yaml [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+              show_icon: false
+              show_label: true
+              label: "[[[ return states[variables.ulm_card_alarm_time_datetime].state ]]]"
+              styles:
+                grid:
+                  - grid-template-areas: "'l'"
+                card:
+                  - background-color: >
+                      [[[
+                        if (!hass.themes.darkMode && variables.mui_card_alarm_time_force_background_color) {
```
Comment (basbruss) [inline custom_cards/custom_card_alarm_time/card_alarm_time.yaml:237]:
```suggestion
                        if (!hass.themes.darkMode && variables.ulm_card_alarm_time_force_background_color) {
```

File: custom_cards/custom_card_alarm_time/card_alarm_time.yaml [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+                      var seconds = "0" + date.getSeconds();
+                      var formattedTime = hours + ':' + minutes.substr(-2) + ':' + seconds.substr(-2);
+                      return formattedTime;
+                    ]]]
+              icon: "mdi:plus"
+              styles:
+                card:
+                  - background-color: >
+                      [[[
+                        if (!hass.themes.darkMode && variables.mui_card_alarm_time_force_background_color) {
```
Comment (basbruss) [inline custom_cards/custom_card_alarm_time/card_alarm_time.yaml:283]:
```suggestion
                        if (!hass.themes.darkMode && variables.ulm_card_alarm_time_force_background_color) {
```

File: custom_cards/custom_card_alarm_time/card_alarm_time.yaml [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+                          if (entity.state != "off") {
+                            return 'rgba(250,250,250,0.8)';
+                          }
+                        }
+                        return 'rgba(var(--color-theme),0.05)';
+                      ]]]
+                icon:
+                  - color: >
+                      [[[
+                        if (!hass.themes.darkMode && variables.mui_card_alarm_time_force_background_color) {
```
Comment (basbruss) [inline custom_cards/custom_card_alarm_time/card_alarm_time.yaml:293]:
```suggestion
                        if (!hass.themes.darkMode && variables.ulm_card_alarm_time_force_background_color) {
```

**Review by basbruss — CHANGES_REQUESTED** [review]

File: custom_cards/custom_card_alarm_time/README.md (hunk 1 above)
Comment (benbur98) [reply custom_cards/custom_card_alarm_time/README.md:27]: Good point, i'd forgotten that.

File: custom_cards/custom_card_alarm_time/card_alarm_time.yaml (hunk 2 above)
Comment (benbur98) [reply custom_cards/custom_card_alarm_time/card_alarm_time.yaml]: Oops, strange copied typo good spot.

**Pull request merged**